aten/src/ATen/native/cuda/BinaryLogicalOpsKernels.cu

```diff
@@ -46,22 +46,74 @@ void logical_and_kernel_cuda(TensorIterator& iter) {
   }
 }
 
+const char logical_or_name[] = "logical_or_kernel";
 void logical_or_kernel_cuda(TensorIterator& iter) {
-  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(kHalf, kBool, ScalarType::BFloat16,
-                                         iter.common_dtype(), "logical_or_cuda", [&]() {
+  auto dtype = iter.common_dtype();
+  if (at::isComplexType(dtype)) {
+#if AT_USE_JITERATOR()
+    static const auto logical_or_string = jiterator_stringify(
+      template <typename T>
+      T logical_or_kernel(T a, T b) {
+        return a || b;
+      }
+    ); // logical_or_string
+    AT_DISPATCH_COMPLEX_TYPES(dtype, "logical_or_cuda", [&]() {
+      jitted_gpu_kernel<
+        /*name=*/ logical_or_name,
+        /*return_dtype=*/ scalar_t,
+        /*common_dtype=*/ scalar_t,
+        /*arity=*/ 2>(iter, logical_or_string);
+    });
+#else
+    AT_DISPATCH_COMPLEX_TYPES(dtype, "logical_or_cuda", [&]() {
+      gpu_kernel_with_scalars(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> bool {
+        return a || b;
+      });
+    });
+#endif
+  } else {
+  AT_DISPATCH_ALL_TYPES_AND3(kHalf, kBool, ScalarType::BFloat16,
+                             dtype, "logical_or_cuda", [&]() {
     gpu_kernel_with_scalars(iter, []GPU_LAMBDA(scalar_t a, scalar_t b) -> bool {
       return a || b;
     });
   });
+  }
 }
 
+const char logical_xor_name[] = "logical_xor_kernel";
 void logical_xor_kernel_cuda(TensorIterator& iter) {
-  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(kHalf, kBool, ScalarType::BFloat16,
-                                         iter.common_dtype(), "logical_xor_cuda", [&]() {
+  auto dtype = iter.common_dtype();
+  if (at::isComplexType(dtype)) {
+#if AT_USE_JITERATOR()
+    static const auto logical_xor_string = jiterator_stringify(
+        template <typename T>
+        T logical_xor_kernel(T a, T b) {
```

Comment:
why is return type T? Also, I don't thing `a != b` is correct, it will result in `true` for different but non-zero a and b/

Reply:
Hi, @ngimel! Yes, I agree with you. made amendments :)

Reply:
The return type here and for logical_or should be `bool`.

See : https://github.com/pytorch/pytorch/blob/148a37682a064eb122db1d5e10b593324cd51d63/aten/src/ATen/native/cuda/BinaryLogicalOpsKernels.cu#L77

Reply:
`bool logical_xor_kernel(T a, T b) { ... }`. Similar for `logical_or`

Reply:
Got that. Thanks, @kshitij12345! I think we will also need to update `logical_and`. Will that be okay?